ddtrace/internal/coverage/code.py

```diff
@@ -10,28 +12,41 @@
 from ddtrace.internal.coverage.report import gen_json_report
 from ddtrace.internal.coverage.report import print_coverage_report
 from ddtrace.internal.coverage.util import collapse_ranges
-from ddtrace.internal.module import BaseModuleWatchdog
+from ddtrace.internal.module import ModuleWatchdog
+from ddtrace.internal.packages import is_stdlib
 from ddtrace.vendor.contextvars import ContextVar
 
 
 _original_exec = exec
 
 ctx_covered = ContextVar("ctx_covered", default=None)
+ctx_is_import_coverage = ContextVar("ctx_is_import_coverage", default=False)
 ctx_coverage_enabled = ContextVar("ctx_coverage_enabled", default=False)
 
 
-class ModuleCodeCollector(BaseModuleWatchdog):
+def _get_ctx_covered_lines() -> t.DefaultDict[str, t.Set]:
+    return ctx_covered.get()[-1] if ctx_coverage_enabled.get() else defaultdict(set)
```

Comment:
The data in the context coverage is a stack (so that nested usage of the context-based collector works properly).

If we've entered the context (which is what sets `ctx_coverage_enabled` to `True` ( at https://github.com/DataDog/dd-trace-py/pull/9609/files#diff-3985b6c4c52757d7d726c3002c85935f582641d687acd0f6ab50f93e1294205dR232-R233 )), then we've already created the stack and added an entry to it.

The only way `ctx_coverage_enabled` becomes `False` is if we `__exit__` the context collector and the stack is empty (at https://github.com/DataDog/dd-trace-py/pull/9609/files#diff-3985b6c4c52757d7d726c3002c85935f582641d687acd0f6ab50f93e1294205dR245-R246 ).